### PR TITLE
Add possibility of fitting on disjoint ranges

### PR DESCRIPTION
**Pull request by fchinu**
With this PR it is possible to define fitting ranges (also on disjoint spaces) using the `limits` keyword. 
For the moment, it is only possible to do so for unbinned fits (as for binned fits we are facing some [issues](https://github.com/zfit/zfit/issues/579)).

This implementation relies on [`TruncatedPDF`](https://zfit.readthedocs.io/en/stable/user_api/pdf/_generated/composed_pdf/zfit.pdf.TruncatedPDF.html#truncatedpdf)s to fit on multiple ranges. A `TruncatedPDF` is now used as default `_total_pdf_` in place of non-truncated `PDF`s, also in case that no restriction of the fit limits is provided.

### CODE REVIEW

File: flarefly/fitter.py [hunk 1]
```diff
@@ -935,7 +948,7 @@ def func_mult(params):
                     floating=not self._fix_bkg_pars_[ipdf]['frac'])
 
         self._total_pdf_ = zfit.pdf.SumPDF(self._signal_pdf_+self._refl_pdf_+self._background_pdf_,
-                                           self._fracs_)
+                                           self._fracs_).to_truncated(limits = self._limits_, obs = obs, norm = obs)
```
Comment (fgrosa) [inline flarefly/fitter.py:951]: Debugging locally I found that the issue with MacOs python3.11 is that this SumPDF is built from the same PDFs as the one used to build the total binned pdf. Since we do not support the truncated PDFs in case of binned fits for the time being, I would suggest the following:
```suggestion
                                           self._fracs_)

        if not self._data_handler_.get_is_binned():
            self._total_pdf_ = self._total_pdf_.to_truncated(limits = self._limits_, obs = obs, norm = obs)
```
combined with the next suggestion.

The tests work in this case, I am not sure if there is any corner case where it breaks though.

File: flarefly/fitter.py [hunk 2]
```diff
@@ -1227,7 +1274,8 @@ def plot_mass_fit(self, **kwargs):
         for signal_pdf in self._signal_pdf_:
             signal_funcs.append(zfit.run(signal_pdf.pdf(x_plot, norm_range=obs)))
         for refl_pdf in self._refl_pdf_:
-            refl_funcs.append(zfit.run(refl_pdf.pdf(x_plot, norm_range=obs)))
+            refl_pdf_pdf = refl_pdf.pdf(x_plot, norm_range=obs)
+            refl_funcs.append(zfit.run(refl_pdf_pdf))
```
Comment (fgrosa) [inline flarefly/fitter.py:1278]:
```suggestion
            refl_funcs.append(zfit.run(refl_pdf.pdf(x_plot, norm_range=obs)))            
```

File: flarefly/fitter.py [hunk 3]
```diff
@@ -207,6 +212,14 @@ def __init__(self, data_handler, name_signal_pdf, name_background_pdf, **kwargs)
             mode=kwargs.get('minuit_mode', 0),
             tol=kwargs.get('tol', 0.001)
         )
+        if 'limits' in kwargs and self._data_handler_.get_is_binned():
+            Logger('Restriction of fit limits is not yet implemented in binned fits!', 'FATAL')
```
Comment (fgrosa) [inline flarefly/fitter.py:216]: this might be included also in the `DataHandler`

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @fchinu, thanks a lot!
A few comments that should fix the failing test with MacOs in case of python3.11

**Review by fgrosa — APPROVED** [review]

**Comment by fchinu** [comment]
Ciao @fgrosa! Thanks a lot for your suggestions! Indeed, the tests now succeed! I have just a question on the addition of the `FATAL` in the `DataHandler` you suggested: I believe that setting a limit on the data is working fine also for the binned case right? Do you want me to add a check for disjoint limits (which I believe are currently not supported in the `DataHandler` implementation we currently have)?

**Comment by fgrosa** [comment]
> Ciao @fgrosa! Thanks a lot for your suggestions! Indeed, the tests now succeed! I have just a question on the addition of the `FATAL` in the `DataHandler` you suggested: I believe that setting a limit on the data is working fine also for the binned case right? Do you want me to add a check for disjoint limits (which I believe are currently not supported in the `DataHandler` implementation we currently have)?

Indeed this suggestion was because of I remembered that we were setting the disjoint limits in the `DataHandler`, but this is not the case, so no need to change anything, thanks!
